test/e2e/storage/volumes.go

```diff
@@ -94,167 +153,6 @@ var _ = utils.SIGDescribe("Volumes", func() {
-
-	Describe("iSCSI [Feature:Volumes]", func() {
-		It("should be mountable", func() {
-			config, _, serverIP := framework.NewISCSIServer(cs, namespace.Name)
-			defer framework.VolumeTestCleanup(f, config)
-
-			tests := []framework.VolumeTest{
-				{
-					Volume: v1.VolumeSource{
-						ISCSI: &v1.ISCSIVolumeSource{
-							TargetPortal: serverIP + ":3260",
-							// from test/images/volumes-tester/iscsi/initiatorname.iscsi
-							IQN:    "iqn.2003-01.org.linux-iscsi.f21.x8664:sn.4b0aae584f7c",
-							Lun:    0,
-							FSType: "ext2",
-						},
-					},
-					File: "index.html",
-					// Must match content of test/images/volumes-tester/iscsi/block.tar.gz
-					ExpectedContent: "Hello from iSCSI",
-				},
-			}
-			fsGroup := int64(1234)
-			framework.TestVolumeClient(cs, config, &fsGroup, tests)
-		})
-	})
-
-	////////////////////////////////////////////////////////////////////////
-	// Ceph RBD
-	////////////////////////////////////////////////////////////////////////
-
-	Describe("Ceph RBD [Feature:Volumes]", func() {
-		It("should be mountable", func() {
-			config, _, secret, serverIP := framework.NewRBDServer(cs, namespace.Name)
-			defer framework.VolumeTestCleanup(f, config)
-			defer cs.CoreV1().Secrets(config.Namespace).Delete(secret.Name, nil)
-
-			tests := []framework.VolumeTest{
-				{
-					Volume: v1.VolumeSource{
-						RBD: &v1.RBDVolumeSource{
-							CephMonitors: []string{serverIP},
-							RBDPool:      "rbd",
-							RBDImage:     "foo",
-							RadosUser:    "admin",
-							SecretRef: &v1.LocalObjectReference{
-								Name: secret.Name,
-							},
-							FSType: "ext2",
-						},
-					},
-					File: "index.html",
-					// Must match content of test/images/volumes-tester/rbd/create_block.sh
-					ExpectedContent: "Hello from RBD",
-				},
-			}
-			fsGroup := int64(1234)
-			framework.TestVolumeClient(cs, config, &fsGroup, tests)
-		})
-	})
-
-	////////////////////////////////////////////////////////////////////////
-	// Ceph
-	////////////////////////////////////////////////////////////////////////
-	Describe("CephFS [Feature:Volumes]", func() {
-		It("should be mountable", func() {
-			config, _, secret, serverIP := framework.NewRBDServer(cs, namespace.Name)
-			defer framework.VolumeTestCleanup(f, config)
-			defer cs.CoreV1().Secrets(config.Namespace).Delete(secret.Name, nil)
-
-			tests := []framework.VolumeTest{
-				{
-					Volume: v1.VolumeSource{
-						CephFS: &v1.CephFSVolumeSource{
-							Monitors:  []string{serverIP + ":6789"},
-							User:      "kube",
-							SecretRef: &v1.LocalObjectReference{Name: secret.Name},
-							ReadOnly:  true,
-						},
-					},
-					File: "index.html",
-					// Must match content of test/images/volumes-tester/ceph/index.html
-					ExpectedContent: "Hello Ceph!",
-				},
-			}
-			framework.TestVolumeClient(cs, config, nil, tests)
-		})
-	})
-
 	////////////////////////////////////////////////////////////////////////
 	// OpenStack Cinder
```

Comment:
Please move OpenStack, GCE PD, vSphere and Azure to the framework too and skip them if it's not running on the right cloud.